src/main/java/duke/tasklist/TaskList.java

```diff
@@ -0,0 +1,152 @@
+     * Verifies if the given taskIndex is valid.
+     * One possible errors are handled. Namely, they are:
+     *      1. taskIndex is out of bound;
+     * @param taskIndex taskIndex from user input after parsed
+     * @return index in int if it is valid
+     * @throws DukeException if invalid index is provided
+     */
+    private static int verifyTaskIndex(int taskIndex) throws DukeException {
+        int index = taskIndex - 1;
+
+        if (index >= tasks.size()) {
+            throw new DukeException("Task with the given index does not exist.");
+        }
+
+        return index;
+    }
+
+    /**
+     * Removes the task with the given index and print the confirmation message.
+     * @param taskIndex taskIndex from user input
+     * @return the corresponding results to be printed to users
+     * @throws DukeException when an invalid taskIndex is entered
+     */
+    public static String[] deleteTask(int taskIndex) throws DukeException {
+        int index = verifyTaskIndex(taskIndex);
+        Task task = tasks.remove(index);
+
+        String[] res = new String[] {
+            "On your command! I have removed this task:",
+            "  " + task.toString(),
+            "Now you have " + tasks.size() + " "
+                + (tasks.size() > 1 ? "tasks" : "task")
+                + " in the list."
+        };
+        return res;
+    }
+
+    /**
+     * Completes the task with the given index and print the confirmation message.
+     *
+     * One possible error is handled. Namely, it is:
+     *      1. the task has been completed;
+     * @param taskIndex taskIndex from user input
+     * @return the corresponding results to be printed to users
+     * @throws DukeException when an invalid taskIndex is entered
+     */
+    public static String[] completeTask(int taskIndex) throws DukeException {
+        int index = verifyTaskIndex(taskIndex);
+        Task task = tasks.get(index);
+
+        if (!task.markAsDone()) {
+            throw new DukeException("Task with the given index has been completed.");
+        }
+
+        String[] res = new String[] {
+            "Wonderful! You have completed this task:",
+            "  " + task.toString()
+        };
+        return res;
+    }
+
+    /**
+     * Adds the task given to the task list
+     * @param newTask a new task to be added
+     * @return the corresponding results to be printed to users
+     */
+    public static String[] addTask(Task newTask) {
+        tasks.add(newTask);
+
+        String[] res = new String[] {
+            "Roger that! Added new task:",
+            " " + newTask.toString(),
+            "Now you have " + tasks.size() + " "
+                + (tasks.size() > 1 ? "tasks" : "task")
+                + " in the list."
+        };
+        return res;
+    }
+
+    /**
+     * Lists the tasks in the task list
+     * @return the information of the tasks present
+     */
+    public static String[] getAllTaskListInfo() {
+        String[] res;
```

Comment:
Perhaps this variable could be given a clearer name? It took me a bit to realize what it was meant to store.

Reply:
Yep sure! I think it is a valid point!